lib/discount.rb

```diff
@@ -0,0 +1,31 @@
+require "./lib/modules/percentage"
+
+class Discount 
+  attr_reader :discount, :description
+  include Percentage
+
+  def initialize options
+    @discount               = options[:discount]
+    @discountable_evaluator = options[:discountable?]
```

Comment:
This again is good, although another way you could do this would be to just allow initialize to take a block which allows for a slightly cleaner interface:

```
Discount.new(discount: '10%') do |amount|
  amount < 0.1
end
```
Then again I think what you've done here might be violating KISS - I'm pretty sure there's a simpler way of doing this - for example if the rule is always just the value is greater than a certain amount, couldn't you just define the minimum amount?

Another way to look at it would be to use inheritance, and have two types of discount that inherit from the base discount class. Sandi Metz talks about this here http://www.neotericdesign.com/blog/2012/4/the-story-of-inheritance